es-rest-client-1.3/core/src/main/java/org/elasticsearch/ElasticSearchExceptionUtil.java

```diff
@@ -0,0 +1,50 @@
+package org.elasticsearch;
+
+import org.elasticsearch.index.Index;
+import org.elasticsearch.indices.RestIndexMissingException;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Responsible for constructing/extracting the proper exception from an ElasticSearch
+ * REST call response when there is an error condition.
+ */
+public final class ElasticSearchExceptionUtil {
+
+    public static final Pattern INDEX_MISSING_EX_PATTERN = Pattern.compile("^IndexMissingException\\[\\[(([^\\]]+))].*");
```

Comment:
After thinking about it more, I think this would be better: 
```
^IndexMissingException\[\[(.*)] missing]$
```
That should completely remove any ambiguity in parsing the message, which should in turn protect us against changes in the api.